include/dem/dem_action_manager.h

```diff
@@ -0,0 +1,429 @@
+#ifndef dem_action_manager_h
+#define dem_action_manager_h
+
+class DEMActionManager
```

Comment:
Maybe needs a description of the class here. I think the usage of a singleton is a good pattern btw, I approve of this message.

Reply:
Thanks!